daffodil-core/src/main/java/org/apache/daffodil/api/Daffodil.java

```diff
@@ -36,4 +49,192 @@ private Daffodil() {
+   * all {@link InfosetOutputter} events
+   *
+   * @return InfosetOutputter
+   */
+  public static InfosetOutputter newNullInfosetOutputter() {
+    return new org.apache.daffodil.runtime1.infoset.NullInfosetOutputter();
+  }
+
+  /**
+   * {@link InfosetInputter} to read an infoset represented as a scala.xml.Node
+   *
+   * @param node the scala.xml.Node infoset
+   * @return InfosetInputter
+   */
+  public static InfosetInputter newScalaXMLInfosetInputter(scala.xml.Node node) {
+    return new org.apache.daffodil.runtime1.infoset.ScalaXMLInfosetInputter(node);
+  }
+
+  /**
+   * {@link InfosetInputter} to read an infoset represented as XML from a java.io.InputStream
+   * <p>
+   * Read in an infoset in the form of XML text from a java.io.InputStream
+   *
+   * @param is the java.io.InputStream to read the XML text from
+   * @return InfosetInputter
+   */
+  public static InfosetInputter newXMLTextInfosetInputter(InputStream is) {
+    return new org.apache.daffodil.runtime1.infoset.XMLTextInfosetInputter(is);
+  }
+
+  /**
+   * {@link InfosetInputter} to read an infoset represented as JSON from a java.io.InputStream
+   * <p>
+   * Read in an infoset in the form of json text from a java.io.InputStream
+   *
+   * @param is the java.io.InputStream to read the json text from
+   * @return InfosetInputter
+   */
+  public static InfosetInputter newJsonInfosetInputter(InputStream is) {
+    return new org.apache.daffodil.runtime1.infoset.JsonInfosetInputter(is);
+  }
+
+  /**
+   * {@link InfosetInputter} to read an infoset represented as an org.jdom2.Document
+   *
+   * @param document the org.jdom2.Document infoset
+   * @return InfosetInputter
+   */
+  public static InfosetInputter newJDOMInfosetInputter(org.jdom2.Document document) {
+    return new org.apache.daffodil.runtime1.infoset.JDOMInfosetInputter(document);
+  }
+
+  /**
+   * {@link InfosetInputter} to read an infoset represented as an org.w3c.dom.Document
+   *
+   * @param document the org.w3c.dom.Document infoset. Note that w3c
+   *                 Documents are not guaranteed to be thread-safe, even if all
+   *                 users only read/traverse it. It is up to the user to ensure
+   *                 that the Document passed into the W3CDOMInfosetInputter is
+   *                 not read or written by other threads while the
+   *                 W3CDOMInfosetInputter has access to it.
+   * @return InfosetInputter
+   */
+  public static InfosetInputter newW3CDOMInfosetInputter(org.w3c.dom.Document document) {
+    return new org.apache.daffodil.runtime1.infoset.W3CDOMInfosetInputter(document);
+  }
+
+  /**
+   * Create an InputSourceDataInputStream from a java.io.InputStream
+   *
+   * @param is input stream to create from
+   * @return InputSourceDataInputStream from a java.io.InputStream
+   */
+  public static InputSourceDataInputStream newInputSourceDataInputStream(InputStream is) {
```

Comment:
These three InputSourceDataInputStream factory methods were the impetus behind this change.

The issue was that previously they lived in the `o.a.d.api.infoset.Infoset` class, which didn't really make sense since they don't really have anything to do with infosets. I'm not sure I love throwing all our factory methods in the `Daffodil` class, but I'm not sure where else makes sense. Does a a new package/class make sense for for ISDIS?

Or, maybe it is actually convenient to say that any Daffodil thing you want to create is either created from a factory method in Daffodil or by something created by one of those factory methods, such that everything ultimately is created from the Daffodil class.

Any thoughts for alternatives?

Reply:
I expect all top level factories we expect people to use to be on the Daffodil object. The finer points of Infosets are really separate from daffodil's core behavior.... that's too subtle for most API users.

Reply:
So do we leave the InfosetInputters/Outputter's in the Infoset object, then? And put the ISDIS in DaffodilIO or just Daffodil?